### PR TITLE
feat: added nextauth with login

### PR DESCRIPTION
**Pull request by pixelass**

### CODE REVIEW

File: types/process.d.ts
Comment (TimPietrusky) [inline types/process.d.ts:1]: Why is this file needed? Where is it actually used?

**Comment by TimPietrusky** [comment]
@doemser I'm not sure if this is desired behavior, but there is something odd with the Login process:

* Stage: Logged out
* Click on Login
* Choose "Login with GitHub"
* I'm not directed to GitHub, as I used the "Login with GitHub" already in the past, I'm logged in
* I see the Avatar in the top right
* I see the "Login with GitHub" button still and I'm not redirected to the page I was before (homepage)


See Screenshot 2023-04-20 at 16 59 35

File: types/process.d.ts
Comment (pixelass) [reply types/process.d.ts:1]: it sets types for our environmental variables so that we don't have to typecast

**Review by TimPietrusky — APPROVED** [review]
LGTM
